### PR TITLE
Make get-x-selection and set-x-selection have configuration choices for clipboard vs. primary selections #673

### PR DESCRIPTION
**Pull request by equwal**
One may desire to use the `clipboard` instead of the `primary` selection, or even clobber both.

Addresses https://github.com/stumpwm/stumpwm/issues/673

### CODE REVIEW

**Comment by xsteve** [comment]
What most applications expect depends on the applications you use.
My experience is that established unix tools use the primary selection.
Only few applications that I use use clipboard.
So I would love to keep primary the default.

One thing that could be done is adding a variable that defaults to primary - you can set it to clipboard.

**Comment by equwal** [comment]
> What most applications expect depends on the applications you use.
> My experience is that established unix tools use the primary selection.
> Only few applications that I use use clipboard.
> So I would love to keep primary the default.
> 
> One thing that could be done is adding a variable that defaults to primary - you can set it to clipboard.

Is there a reason why we should not clobber both? I can do that also.

**Comment by dmb2** [comment]
Hi, I see you've made a number of really great contributions recently.  One thing I try hard to do (which constrains what makes it into the main stumpwm branch) is to avoid opinionated defaults.  In this case, the current behavior is as close to the standard as we can be.  At times its inconvenient an not intuitive, but its the best we've got.

I like the idea of making this configurable.  Could you set the default to the old behavior and add the new variables to the manual?

**Comment by equwal** [comment]
> Hi, I see you've made a number of really great contributions recently. One thing I try hard to do (which constrains what makes it into the main stumpwm branch) is to avoid opinionated defaults. In this case, the current behavior is as close to the standard as we can be. 

>At times its inconvenient an not intuitive, but its the best we've got.
> I like the idea of making this configurable. Could you set the default to the old behavior and add the new variables to the manual?

This is a fair point. Arbitrarily clobbering two selection spaces by default is kind of strange. On the other hand, I wonder at where my selections are going sometimes.

File: selection.lisp [hunk 1]
```diff
@@ -25,7 +25,13 @@
...
 
 (export '(get-x-selection
-          set-x-selection))
+          set-x-selection
+          *default-selections*))
+
+(defvar *default-selections* '(:primary)
+  #.(format nil "~@{~A~}"
+          "Either a list of keyword X selections (like :primary or :clipboard)"
+          ", or a single one."))
```
Comment (dmb2) [inline selection.lisp]: Please document all possible values and what they do.

File: selection.lisp [hunk 2]
```diff
@@ -102,24 +106,25 @@
...
-          (xlib:convert-selection selection :utf8_string (screen-input-window (current-screen)) :stumpwm-selection)
-          ;; Note: this may spend longer than timeout in this loop but it will eventually return.
-          (let ((time (get-internal-real-time)))
-            (loop for ret = (xlib:process-event *display* :handler #'wait-for-selection :timeout timeout :discard-p nil)
-                  when (or ret
-                           (> (/ (- time (get-internal-real-time)) internal-time-units-per-second)
-                              timeout))
-                  ;; make sure we return a string
-                  return (or ret "")))))))
+  (let ((selection (if (listp selection) (car selection) selection)))
```
Comment (equwal) [inline selection.lisp]: #L109 is the only non-whitespace change.

**Comment by dmb2** [comment]
Good to go once you fix merge conflicts.

**Pull request merged**